cl_manycore/libraries/bsg_manycore_eva.cpp

```diff
@@ -294,8 +391,87 @@ static uint32_t default_get_dram_x_shift(const hb_mc_manycore_t *mc)
+ * (i.e. stripe size = dram bank size = 0x800_0000)
+ *
+ * DRAM EPA  =  EPA_top + block_offset + word_addressible  
+ * DRAM NPA  =  <Y coord, X coord, DRAM EPA>
  */
 static int default_eva_to_npa_dram(const hb_mc_manycore_t *mc,
+                                   const hb_mc_coordinate_t *o,
+                                   const hb_mc_coordinate_t *src,
+                                   const hb_mc_eva_t *eva,
+                                   hb_mc_npa_t *npa,
+                                   size_t *sz)
+{
+        int rc;
+        const hb_mc_config_t *cfg = hb_mc_manycore_get_config(mc);
+        hb_mc_idx_t x,y;
+        hb_mc_epa_t epa;
+
+        // Calculate X coordinate of NPA from EVA
+        rc = default_eva_get_x_coord_dram (mc, cfg, eva, &x); 
+        if (rc != HB_MC_SUCCESS) { 
+                bsg_pr_err("%s: failed to generate x coordinate from eva 0x%08" PRIx32 ".\n",
+                           __func__,
+                           hb_mc_eva_addr(eva));
+                return rc;
+        }
+
+        // Calculate Y coordinate of NPA from EVA
+        rc = default_eva_get_y_coord_dram (mc, cfg, eva, &y);
+        if (rc != HB_MC_SUCCESS) { 
+                bsg_pr_err("%s: failed to generate y coordinate from eva 0x%08" PRIx32 ".\n",
+                           __func__,
+                           hb_mc_eva_addr(eva));
+                return rc;
+        }
+
+
+        // Calculate EPA Portion of NPA from EVA
+        rc = default_eva_get_epa_dram (mc, cfg, eva, &epa, sz);
+        if (rc != HB_MC_SUCCESS) { 
+                bsg_pr_err("%s: failed to generate npa from eva 0x%08" PRIx32 ".\n",
+                           __func__,
+                           hb_mc_eva_addr(eva));
+                return rc;
+        }
+
+	*npa = hb_mc_epa_to_npa(hb_mc_coordinate(x,y), epa);
+
+	bsg_pr_dbg("%s: Translating EVA 0x%08" PRIx32 " for tile (x: %d y: %d) to NPA {x: %d y: %d, EPA: 0x%08" PRIx32 "} sz = %08x. \n",
+		   __func__, hb_mc_eva_addr(eva),
+		   hb_mc_coordinate_get_x(*src),
+		   hb_mc_coordinate_get_y(*src),
+		   hb_mc_npa_get_x(npa),
+		   hb_mc_npa_get_y(npa),
+		   hb_mc_npa_get_epa(npa),
+		   uint32_t(*sz));
+
+	return HB_MC_SUCCESS;
+}
+
+ 
+/**
+ * Converts a DRAM Endpoint Virtual Address to a Network Physical Address and
+ * size (contiguous bytes following the specified EVA)
+ * @param[in]  cfg    An initialized manycore configuration struct
+ * @param[in]  o      Coordinate of the origin for this tile's group
+ * @param[in]  src    Coordinate of the tile issuing this #eva
+ * @param[in]  eva    An eva to translate
+ * @param[out] npa    An npa to be set by translating #eva
+ * @param[out] sz     The size in bytes of the NPA segment for the #eva
+ * @return HB_MC_FAIL if an error occured. HB_MC_SUCCESS otherwise.
+ */
+__attribute__((deprecated))
+static int default_eva_to_npa_dram_dep(const hb_mc_manycore_t *mc,
```

Comment:
Why is this still being used if it's deprecated and static?

Reply:
It's not being used anywhere, but Michael insisted on not removing the old API for now, so I left it in with a deprecated tag.